### PR TITLE
MAINT: remove equality constraint that was impossible to satisfy with…

### LINKED ISSUE #21760: BUG: failure in `scipy/optimize/tests/test_minimize_constrained.py::test_gh11649`
### Describe your issue.

There is a test failure in main causing most CI jobs to fail, e.g. see gh-21744.

### Reproducing Code Example

```python
-
```

### Error message

```shell
================================== FAILURES ===================================
________________________________ test_gh11649 _________________________________

    def test_gh11649():
        bnds = Bounds(lb=[-1, -1], ub=[1, 1], keep_feasible=True)
    
        def assert_inbounds(x):
            assert np.all(x >= bnds.lb)
            assert np.all(x <= bnds.ub)
    
        def obj(x):
            assert_inbounds(x)
            return np.exp(x[0])*(4*x[0]**2 + 2*x[1]**2 + 4*x[0]*x[1] + 2*x[1] + 1)
    
        def nce(x):
            assert_inbounds(x)
            return x[0]**2 + x[1]
    
        def nce_jac(x):
            return np.array([2*x[0], 1])
    
        def nci(x):
            assert_inbounds(x)
            return x[0]*x[1]
    
        x0 = np.array((0.99, -0.99))
        nlcs = [NonlinearConstraint(nci, -10, np.inf),
                NonlinearConstraint(nce, 1, 1, jac=nce_jac)]
    
        res = minimize(fun=obj, x0=x0, method='trust-constr',
                       bounds=bnds, constraints=nlcs)
>       assert res.success
E       AssertionError: assert False
E        +  where False =            message: Constraint violation exceeds 'gtol'\n           success: False\n            status: 4\n              ...enalty: 1.0\n barrier_parameter: 2.048000000000001e-09\n barrier_tolerance: 2.048000000000001e-09\n             niter: 225.success

C:\hostedtoolcache\windows\Python\3.11.9\x64\Lib\site-packages\scipy\optimize\tests\test_minimize_constrained.py:741: AssertionError
```

### SciPy/NumPy/Python version and system information

```shell
main
```

### PR DESCRIPTION
**Pull request by andyfaff**
closes #21760.

Not sure this is the best solution. The existing test was nowhere near the true solution and wasn't satisfying constraints, whilst still asserting that the minimize succeeded.

### CODE REVIEW

File: scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py [hunk 1]
```diff
@@ -302,6 +302,10 @@ def _minimize_trustregion_constr(fun, x0, args, grad,
         * 3 : `callback` function requested termination.
         * 4 : Constraint violation exceeds 'gtol'.
 
+        .. versionchanged:: 1.15.0
+            If the constraint violation exceeds gtol then `result.success`
+            will be False. Previously this could've been reported as True.
```
Comment (mdhaber) [inline scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py:307]:
```suggestion
            If the constraint violation exceeds `gtol`, then `result.success`
            will now be False.
```

**Review by mdhaber — COMMENTED** [review]
Do I understand that the problem has an optimal solution and `trust-constr` is not finding it? Maybe that is a separate issue.

> Not sure this is the best solution. 

I see that the purpose of the test was just to make sure that the bounds are respected. Did you consider just commenting out `assert res.success` (and maybe we change gh-21760 to be about `trust-constr` failing to find the optimal solution of this problem)?

> Previously this could've been reported as True.

Contractions are not super common in the documentation. How about:
```
If the constraint violation exceeds gtol, then `result.success` will now be False.
```
The addition of "now" implies that this was not the case before.

**Comment by andyfaff** [comment]
> Do I understand that the problem has an optimal solution and trust-constr is not finding it

That is correct. I'm not keyed up on this area of the code-base, or the solver, to know if this is an issue or not. My preference would be status quo until someone highlights that it's a real performance problem.

I've made the suggested changes to #21760.

File: scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py (hunk 1 above)
Comment (andyfaff) [reply scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py]: Made the commit separately (saw this afterwards)

**Comment by mdhaber** [comment]
CircleCI received a "kill" signal, and there was a slow test, but seems safe to merge. Thanks @andyfaff!

**Pull request merged**